csrc/optimization/alias_analysis.cpp

```diff
@@ -247,6 +248,81 @@ void AliasFinder::handle(const LoadStoreOp* permute) {
   analysis_.add(out, in, std::move(out_layout));
 }
 
+void AliasFinder::handle(const SliceOp* slice) {
+  TensorView* in = slice->in();
+  TensorView* out = slice->out();
+
+  const std::vector<IterDomain*>& in_rfactor = in->getMaybeRFactorDomain();
+  const std::vector<IterDomain*>& out_root = out->getRootDomain();
+  const std::vector<IterDomain*>& out_rfactor = out->getMaybeRFactorDomain();
+
+  std::unordered_map<IterDomain*, IterDomain*> in_rfactor_to_out_root =
+      PairwiseRootDomainMap(in, out).mapBroadcast(true).mapProducerToConsumer();
+
+  const auto out_rank = out_rfactor.size();
+  std::unordered_map<IterDomain*, IterDomain*> out_root_to_rfactor;
+  out_root_to_rfactor.reserve(out_rank);
+  for (auto i : c10::irange(out_rank)) {
+    out_root_to_rfactor[out_root[i]] = out_rfactor[i];
+  }
+
+  Layout in_layout = analysis_.preferredLayout(in);
+  if (!ir_utils::computePermutation(in_rfactor, in_layout.allocation_domain)
+           .has_value()) {
+    // Give up when `in`'s allocation domain is not an rfactor permutation.
+    return;
+  }
+
+  // Inherit the allocation order from the input.  However, refine the
+  // contiguity flags.
+  Layout out_layout;
+  out_layout.allocation_domain.reserve(out_rank);
+  for (IterDomain* in_allocation_id : in_layout.allocation_domain) {
+    if (!in_rfactor_to_out_root.count(in_allocation_id)) {
+      // `in_allocation_id` is a reduction product.
+      continue;
+    }
+    IterDomain* out_root_id = in_rfactor_to_out_root.at(in_allocation_id);
+    out_layout.allocation_domain.push_back(out_root_to_rfactor.at(out_root_id));
+  }
+
+  // Scan through the allocation domain in minor-to-major order. If an
+  // IterDomain is sliced, the next non-broadcast IterDomain has to be marked
```

Comment:
Agree with Jacob. i.e. if we later decide to merge a trivially sliced dimension, we should recognize that and merge it back I guess.

But this can only be done opportunistically (only handle static trivial slice), I guess?! Since padding value can be a runtime value and we can't know for sure that it'll be a trivial slice or not until runtime.

Reply:
> Is it possible to encounter a trivial SliceOp here? Where both left and right pad are zero? We should treat such case as the identity on that axis.

https://github.com/NVIDIA/Fuser/blob/2e2f3c7c3c07545cd673ac95174d124892a520e7/csrc/ops/alias.cpp#L792 and https://github.com/NVIDIA/Fuser/blob/2e2f3c7c3c07545cd673ac95174d124892a520e7/csrc/ops/alias.cpp#L772 have already done a good job filtering out trivial slices and trivial slice dimensions. So I'd rather keep the logic here simple.